setuptools/glob.py

```diff
@@ -92,8 +90,8 @@ def _iglob(pathname, recursive):
 
 def glob1(dirname, pattern):
     if not dirname:
-        if isinstance(pattern, binary_type):
-            dirname = os.curdir.encode('ASCII')
+        if isinstance(pattern, bytes):
+            dirname = os.curdir.decode('ASCII')
```

Comment:
So while I think you're right about this change (encode->decode), it should probably be in a separate commit and maybe even a separate PR, as it's very different from the stated intention of the commit.

Reply:
I think this is wrong: the goal of `glob` is to support this usage too:
```python
python -c 'from setuptools.glob import glob; print(glob(b"*.rst"))'
```
no?
in any case, this fails with this PR.

Reply:
On `master`, from setuptools directory, the results are:
```python
[b'README.rst', b'towncrier_template.rst', b'CHANGES.rst']
```

Reply:
This is probably a good time to add some tests.

Reply:
It looks like the intent of the patch was to change to:
```python
dirname = os.curdir.encode('ASCII')
```